mantis-common/src/test/java/io/mantisrx/common/compression/CompressionUtilsTest.java

```diff
@@ -67,6 +67,61 @@
         }
     }
 
+    @Test public void testDelimiterWiithPrefixMatchingEndOfMEssage() {
+        // Delimiter starts with 'c', event1 ends with 'c'
+        String delimiter = "ccd";
+
+        String event1 = "abc";
+        String event2 = "def";
+        String event3 = "ghi";
+        String testInput = event1
+                + delimiter
+                + event2
+                + delimiter
+                + event3;
+        try (BufferedReader reader = new BufferedReader(new StringReader(testInput))) {
+            List<MantisServerSentEvent> result = CompressionUtils.tokenize(reader, delimiter);
+
+            assertEquals("Delimiter: '" + delimiter + "'", result.size(), 3);
+            assertEquals(event1, result.get(0).getEventAsString());
+            assertEquals(event2, result.get(1).getEventAsString());
+            assertEquals(event3, result.get(2).getEventAsString());
+
+        } catch (IOException ex) {
```

Comment:
I don't think it's necessary to catch this exception here. Just make the test method throw exception, and the test will simply fail in case of IOE. Same in the other test.